lib/oelite/task.py

```diff
@@ -209,46 +202,61 @@ def run(self):
             os.remove(self.logsymlink)
         os.symlink(os.path.basename(self.logfn), self.logsymlink)
 
-        real_stdin = os.dup(sys.stdin.fileno())
-        real_stdout = os.dup(sys.stdout.fileno())
-        real_stderr = os.dup(sys.stderr.fileno())
-        os.dup2(stdin.fileno(), sys.stdin.fileno())
-        os.dup2(logfile.fileno(), sys.stdout.fileno())
-        os.dup2(logfile.fileno(), sys.stderr.fileno())
+    def save_context(self):
+        self.old_stdin = os.dup(sys.stdin.fileno())
+        self.old_stdout = os.dup(sys.stdout.fileno())
+        self.old_stderr = os.dup(sys.stderr.fileno())
```

Comment:
I know you did not implement this, but shouldn't most of this code handle OSError exceptions?